### ISSUE TITLE
HCA 837 Extract - need to create a "catch up extract"

### ISSUE THREAD

**Issue by stratabuild** [stratasan/wrike-test#139]
**Product Owner: Christian**
**Context:** 
2020-06-26 - Sent HCA a test with a full 3 TB extract of APCD patient detail. This included data from Q4 2016 - May 2020
2020-07 Sent HCA an incremental extract for them to test loading an incremental with a full
**The Problem we are trying to solve**
10/22 - HCA no longer wants a full extract but a gap from the end of the provided extract to today.
Joseph Van Hook from HCA provided a request:
We are exploring a new approach to our production move. Instead of loading a new full historical file with 13 quarters, we would like an incremental file(s). This incremental file would contain any updates or new records from the time the previous file was generated.
 Do you think this is something you guys can accommodate? We are looking for a mid-November go-live at this time and can adjust based on this ask."

**Assumptions/ Risks**

* We are able to identify updates to previous records that go back 13 quarters
* We are able to provide new records that were not previously provided

**Desired Outcomes**

* HCA wants to get their full APCD data up to date without getting a new 3 TB Extract

**Desired Output**

* Options on how we could implement to present to HCA

┆Issue is synchronized with this [Wrike Task](https://www.wrike.com/open.htm?id=582884944)

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

Jackie DeGroatAlex Antonison who provided them a date of Mid November?

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

Do we do this free or is there a cost?

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

Who is this request for?
HCA

What is the request?
HCA encountered some issues loading the full extract we provided them in June. As such, for their go-live instead of us sending them another full extract as planned, they would like for us to send them a "catch up" extract that would bring their warehouse up to date from the last extract we sent them on 2020-06-26.
This is not something we regularly do and will require some thought as to how to accomplish this.

Why is the request needed?
HCA struggled to process the initial full extract we sent them and would like for us to instead catch them up from the extract we sent them in June.

How often do you hear this request?
This is the first time
When is this request needed by?
Urgent - within 4 weeks
Request For
Data (APCD, State, Other)
Links to queries/reports
Urgent Request - Need w/in 4 weeks
Why is this an urgent request?
HCA is wanting a mid-November go live. As such it would require us to start working on it in early November.
Since this is an urgent request, please alert the product team about it in the #product slack channel. Did you do that?
Yes
What date is this needed by?
2020-11-19

**Comment by stratabuild** [comment]
➤ Alex Antonison commented:

Christian Lawrence The mid-November timeline was a request from HCA as that is when they are wanting to launch.  No promises have been made around timeline.  The  I have cc'ed you on the e-mails with HCA indicating that I am working with Product to evaluate when we can do this work.

I arbitrarily selected 2020-11-19 as that was a "mid" November date.  If we need to adjust that we just need to manage HCA's expectations.

I do not believe it is a large amount enough of work to additionally bill them.  To provide context this is a part of the HCA 837 Extract upsell sold to them around June this year.

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

From Alex: This request is as a result of technical issues that have had on their side processing the "full" extract we sent them. Originally after we sent them the initial "test" full extract, we were going to send them a new "full" extract for go live....however they struggled to even process the "test" full extract (which when I say test it is more on their end, we sent them an entirely valid full extract).
What they are wanting us to do instead of sending a new full extract is to send them a "catch up" incremental file that will bring the extract we sent them in June to be current.....unfortunately we don't manage our data that way so we would need to figure out how we would accomplish this

**Comment by stratabuild** [comment]
➤ Christian commented:

@alex-strat are you able to provide implementation options and proposed dates so we can provide to HCA?

**Comment by stratabuild** [comment]
➤ alex-strat commented:

clawdyne That is what is going to require work.  This would need to be added to the sprint so I can do the research to give the options + dates

**Comment by stratabuild** [comment]
➤ alex-strat commented:

drs1832 Here is the ETL load spec that includes all of the details of where the data was loaded and the settings for the unloaded files - https://docs.google.com/document/d/1HGgXxWlN2_rqBA-tS_UKjtlO1t3JzCyzEJ_LcULZcQo/edit

You will more than likely need to use the peach IAM role that I will send you via Slack.

**Comment by stratabuild** [comment]
➤ Christian commented:

@alex-strat what are our options / target dates?

**Comment by stratabuild** [comment]
➤ alex-strat commented:

clawdyne Dave was able to verify our initial theory that we can produce a "catch up" extract that would include all claims in our allpayer_patient_claim_detail_view that have an extract date that goes beyond the initial extract date of our initial large extract.

One unknown that I have reached out to HCA about is whether or not they have processed the incremental extract we sent them or not.  That would simply determine whether we include or do not include it in the catch-up extract.

Another realization that came out of this was that there was a historical load provided from Vendormort that we loaded in July as a part of adding a new clearinghouse which would require us to send them two "catch up" extracts.

  * One where we send them the historical load that we received on 7/13 to be processed first
  * A second which would be all updated claims between July and October extracts (depending on if they processed the incremental we sent them or not)

I could see this ticket for creating this to be about a 3 because we are having to run it all manually because it wouldn't make sense to adjust our production code for this edge case.  I would like to hold out until HCA responds with whether or not they processed the incremental prior to starting work but we could go ahead and create a ticket.  With this not being a large amount of work plus they are good with an early December delivery, I think we could start work on it in a week or so with plenty of time to complete by early December.
